alodataset/kitti_stereo_flow2012.py

```diff
@@ -0,0 +1,244 @@
+import os
+import cv2
+import torch
+import numpy as np
+from typing import Dict
+
+from alodataset import BaseDataset, Split, SplitMixin
+from aloscene import Frame, Disparity, Mask, Flow
+from aloscene.camera_calib import CameraIntrinsic, CameraExtrinsic
+
+
+class KittiStereoFlow2012(BaseDataset, SplitMixin):
+    SPLIT_FOLDERS = {Split.TRAIN: "training", Split.TEST: "testing"}
+
+    def __init__(
+        self,
+        name="kitti_stereo2012",
+        grayscale=False,
+        sequence_start=10,
+        sequence_end=11,
+        load: list = [
+            "disp_noc",
+            "disp_occ",
+            "disp_refl_noc",
+            "disp_refl_occ",
+            "flow_occ",
+            "flow_noc",
+        ],
+        **kwargs,
+    ):
+        super().__init__(name=name, **kwargs)
+        self.grayscale = grayscale
+        self.load = load
+        self.sequence_start = sequence_start
+        self.sequence_end = sequence_end
+
+        if self.sample:
+            return
+
+        assert sequence_start <= sequence_end, "sequence_start should be less than sequence_end"
+
+        if "disp_noc" in load or "disp_occ" in load or "disp_refl_noc" in load or "disp_refl_occ" in load:
+            assert sequence_start <= 11 and sequence_end >= 10, "Disparity is not available for this frame range"
+        if "flow_occ" in load or "flow_noc" in load:
+            assert sequence_start <= 10 and sequence_end >= 10, "Flow is not available for this frame range"
+
```

Comment:
Ground-truth is computed for the gray camera. If ground-truth (disp or flow) is asked, make sure `grayscale==True`